src/Subscription/Engine/CatchUpSubscriptionEngine.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Patchlevel\EventSourcing\Subscription\Engine;
+
+use Patchlevel\EventSourcing\Subscription\Subscription;
+
+use const PHP_INT_MAX;
+
+final class CatchUpSubscriptionEngine implements SubscriptionEngine
+{
+    public function __construct(
+        private readonly SubscriptionEngine $parent,
+        private readonly int|null $limit = null,
```

Comment:
Maybe instead of `null` add here `INT_MAX` as default value?

Reply:
I had that before and changed it to `null`. The reason is I would also have to set it as a default value in Bundle. And if we decide to add a third default parameter (e.g. logger) and the developer only wants to set this, he must also set this constant.

Regarding the constant itself: This is an implementation detail. I took the constant because it works internally with a `for` loop. If the code is rewritten internally, it may also be that the constant is no longer needed.